### PR TITLE
Add DataTable::appendColumn()

### PR DESCRIPTION
**Pull request by klshrinidhi**
Addressing #1578. 

This PR adds method `DataTable::appendColumn()`.

### CODE REVIEW

File: OpenSim/Common/DataTable.h [hunk 1]
```diff
@@ -758,6 +760,137 @@ class DataTable_ : public AbstractDataTable {
...
+    \endcode
+
+    \param columnLabel Label of the column to be added. Must not be same as the
+                       label of an existing column.
+    \param container Sequence container holding the elements of the column to be
+                     appended.
+    \throws InvalidCall If DataTable_ contains no rows at the time of this call.
+    \throws InvalidArgument If columnLabel specified already exists in the
+                            DataTable_.
+    \throws InvalidColumn If the input column contaiins incorrect number of
```
Comment (chrisdembia) [inline OpenSim/Common/DataTable.h]: Typo "contaiins" (see below as well).

File: OpenSim/Common/AbstractDataTable.h [hunk 2]
```diff
@@ -56,6 +61,21 @@ class IncorrectNumColumns : public InvalidRow {
...
+class IncorrectNumRows : public InvalidColumn {
+public:
+    IncorrectNumRows(const std::string& file,
+                     size_t line,
+                     const std::string& func,
+                     size_t expected,
+                     size_t received) :
+        InvalidColumn(file, line, func) {
+        std::string msg = "expected = " + std::to_string(expected);
+        msg += " received = " + std::to_string(received);
```
Comment (chrisdembia) [inline OpenSim/Common/AbstractDataTable.h]: Will this message say "Incorrect number of rows"? If not, it'll be unclear what the numbers are.

Comment (tkuchida) [inline OpenSim/Common/AbstractDataTable.h]: The message would also be easier to interpret with a comma (or other non-space delimiter) before "received".

File: OpenSim/Common/DataTable.h [hunk 3]
```diff
@@ -758,6 +760,137 @@ class DataTable_ : public AbstractDataTable {
...
+
+        Vector col{static_cast<int>(std::distance(begin, end))};
+        int ind{0};
+        for(auto it = begin; it != end; ++it)
+            col[ind++] = *it;
+
+        appendColumn(columnLabel, col);
+    }
+
+    /** Append column to the DataTable_ using a SimTK::Vector_.
```
Comment (tkuchida) [inline OpenSim/Common/DataTable.h]: `Vector_` -> `Vector`?

File: OpenSim/Common/DataTable.h [hunk 4]
```diff
@@ -758,6 +760,137 @@ class DataTable_ : public AbstractDataTable {
...
+    void appendColumn(const std::string& columnLabel,
+                      const Vector& depCol) {
+        appendColumn(columnLabel, depCol.getAsVectorView());
+    }
+
+    /** Append column to the DataTable_ using a SimTK::VectorView.
+
+    \param columnLabel Label of the column to be added. Must not be same as the
+                       label of an existing column.
+    \param container Sequence container holding the elements of the column to be
```
Comment (tkuchida) [inline OpenSim/Common/DataTable.h]: This method has no such argument.

File: OpenSim/Common/DataTable.h [hunk 5]
```diff
@@ -758,6 +760,137 @@ class DataTable_ : public AbstractDataTable {
...
+    std::vector<double> col{};
+    // ......
+    // Fill up 'col'.
+    // ......
+    table.append("new-column", col.begin(), col.end());
+    \endcode
+
+    \param columnLabel Label of the column to be added. Must not be same as the
+                       label of an existing column.
+    \param container Sequence container holding the elements of the column to be
```
Comment (tkuchida) [inline OpenSim/Common/DataTable.h]: This method has no such argument.

File: OpenSim/Common/DataTable.h [hunk 6]
```diff
@@ -758,6 +760,137 @@ class DataTable_ : public AbstractDataTable {
...
+            col[ind++] = *it;
+
+        appendColumn(columnLabel, col);
+    }
+
+    /** Append column to the DataTable_ using a SimTK::Vector_.
+
+    \param columnLabel Label of the column to be added. Must not be same as the
+                       label of an existing column.
+    \param container Sequence container holding the elements of the column to be
```
Comment (tkuchida) [inline OpenSim/Common/DataTable.h]: This method has no such argument.

File: OpenSim/Common/DataTable.h [hunk 7]
```diff
@@ -758,6 +760,137 @@ class DataTable_ : public AbstractDataTable {
...
+    \throws InvalidCall If DataTable_ contains no rows at the time of this call.
+    \throws InvalidArgument If columnLabel specified already exists in the
+                            DataTable_.
+    \throws InvalidColumn If the input column contaiins incorrect number of 
+                          rows.                                               */
+    void appendColumn(const std::string& columnLabel,
+                      const VectorView& depCol) {
+        OPENSIM_THROW_IF(getNumRows() == 0,
+                         InvalidCall,
+                         "DataTable must have one or more rows before we can"
```
Comment (tkuchida) [inline OpenSim/Common/DataTable.h]: Will read "canappend".

File: OpenSim/Common/DataTable.h (hunk 1 above)
Comment (klshrinidhi) [inline OpenSim/Common/DataTable.h]: Done.

File: OpenSim/Common/AbstractDataTable.h (hunk 2 above)
Comment (klshrinidhi) [inline OpenSim/Common/AbstractDataTable.h]: Done.

File: OpenSim/Common/DataTable.h (hunk 5 above)
Comment (klshrinidhi) [inline OpenSim/Common/DataTable.h]: Fixed.

File: OpenSim/Common/DataTable.h (hunk 3 above)
Comment (klshrinidhi) [inline OpenSim/Common/DataTable.h]: Done.

File: OpenSim/Common/DataTable.h (hunk 6 above)
Comment (klshrinidhi) [inline OpenSim/Common/DataTable.h]: Fixed.

File: OpenSim/Common/DataTable.h (hunk 4 above)
Comment (klshrinidhi) [inline OpenSim/Common/DataTable.h]: Fixed.

File: OpenSim/Common/DataTable.h (hunk 7 above)
Comment (klshrinidhi) [inline OpenSim/Common/DataTable.h]: Fixed.

File: OpenSim/Common/DataTable.h (hunk 1 above)
Comment (chrisdembia) [inline OpenSim/Common/DataTable.h]: Thank you; did you fix the other occurrences of this typo?

File: OpenSim/Common/DataTable.h [hunk 8]
```diff
@@ -758,6 +760,138 @@ class DataTable_ : public AbstractDataTable {
...
+
+    \param columnLabel Label of the column to be added. Must not be same as the
+                       label of an existing column.
+    \param begin Iterator referring to the beginning of the range.
+    \param end Iterator referring to the end of the range.
+
+    \throws InvalidCall If DataTable_ contains no rows at the time of this call.
+    \throws InvalidArgument If columnLabel specified already exists in the
+                            DataTable_.
+    \throws InvalidColumn If the input column contaiins incorrect number of
```
Comment (tkuchida) [inline OpenSim/Common/DataTable.h]: contaiins -> contains

File: OpenSim/Common/DataTable.h [hunk 9]
```diff
@@ -758,6 +760,138 @@ class DataTable_ : public AbstractDataTable {
...
+    /** Append column to the DataTable_ using a SimTK::Vector.
+
+    \param columnLabel Label of the column to be added. Must not be same as the
+                       label of an existing column.
+    \param depCol Column vector to be appended to the table.
+
+    \throws InvalidCall If DataTable_ contains no rows at the time of this call.
+    \throws InvalidArgument If columnLabel specified already exists in the
+                            DataTable_.
+    \throws InvalidColumn If the input column contaiins incorrect number of
```
Comment (tkuchida) [inline OpenSim/Common/DataTable.h]: contaiins -> contains

File: OpenSim/Common/DataTable.h [hunk 10]
```diff
@@ -758,6 +760,138 @@ class DataTable_ : public AbstractDataTable {
...
+    /** Append column to the DataTable_ using a SimTK::VectorView.
+
+    \param columnLabel Label of the column to be added. Must not be same as the
+                       label of an existing column.
+    \param depCol Column vector to be appended to the table.
+
+    \throws InvalidCall If DataTable_ contains no rows at the time of this call.
+    \throws InvalidArgument If columnLabel specified already exists in the
+                            DataTable_.
+    \throws InvalidColumn If the input column contaiins incorrect number of
```
Comment (tkuchida) [inline OpenSim/Common/DataTable.h]: contaiins -> contains

**Comment by tkuchida** [comment]
:+1: LGTM. Travis failure is familiar timeout; merging :car::dash:

**Pull request merged**